pyqtorch/utils.py

```diff
@@ -114,5 +115,49 @@ def _rand(n_qubits: int) -> torch.Tensor:
     return _state.reshape([2] * n_qubits + [batch_size]).to(device=device)
 
 
-def param_dict(keys: Sequence[str], values: Sequence[torch.Tensor]) -> dict[str, torch.Tensor]:
+def param_dict(keys: Sequence[str], values: Sequence[Tensor]) -> dict[str, Tensor]:
     return {key: val for key, val in zip(keys, values)}
+
+
+def density_mat(state: Tensor) -> Tensor:
+    """
+    Computes the density matrix from a pure state vector.
+
+    Args:
+        state (Tensor): The pure state vector :math:`|\\psi\\rangle`.
+
+    Returns:
+        Tensor: The density matrix :math:`\\rho = |\psi \\rangle \\langle\\psi|`.
+
+    Raises:
+        TypeError: If the input is not a Tensor.
+            The input must be a Tensor.
+    """
+
+    # Verification input type:
+    if not isinstance(state, Tensor):
+        raise TypeError("The input must be a Tensor")
+
+    n_qubits = len(state.size()) - 1
+
+    # Permute the vector's batch_first
+    batch_dim = state.dim() - 1
+    batch_first_perm = [batch_dim] + list(range(batch_dim))
+    state = torch.permute(state, batch_first_perm)
+
+    # reshape by flatten
+    state = state.flatten()
+
+    # Split for every batch
+    ket_list = torch.split(state, split_size_or_sections=2**n_qubits)
```

Comment:
`torch.split` returns a `Tuple[Tensor]`: https://pytorch.org/docs/stable/generated/torch.split.html.
So I will change it because it's not a `List`.